tests/test_app.py

```diff
@@ -6,45 +6,90 @@
 
 import pytest
+from telliot.utils.app import AppConfig
 from telliot.utils.app import Application
 from telliot.utils.app import default_homedir
 
 
+# Temp file location
+testhome = Path("./temp").resolve().absolute()
+
+
 def test_homedir():
     """Test default home directory"""
     hd = default_homedir()
     assert isinstance(hd, pathlib.Path)
     assert hd.exists()
 
 
-def test_application():
-    """Test application features"""
-
-    testhome = "./tmpapp"
-    if pathlib.Path(testhome).exists():
+def test_application_homedir():
+    """Test home directory handling"""
+    if testhome.exists():
         shutil.rmtree(testhome)
 
-    #: Create an application with the default home directory
-    app = Application(name="appname")
-    assert app.homedir == default_homedir()
-
     with pytest.raises(FileExistsError):
-        app = Application(name="appname", homedir=testhome)
+        app = Application(name="appname", homedir=testhome, config_class=AppConfig)
 
-    os.mkdir(testhome)
-    app = Application(name="appname", homedir=testhome)
+    testhome.mkdir(parents=True)
+    app = Application(name="appname", homedir=testhome, config_class=AppConfig)
     assert isinstance(app.homedir, pathlib.Path)
 
     shutil.rmtree(testhome)
 
 
-def test_application_proc():
+def test_application_default_home():
+    """Test default appliaction directory"""
+
+    app = Application(name="appname", config_class=AppConfig)
+    assert app.homedir == default_homedir()
+    assert "telliot" in str(app.homedir)
+
+
+def test_application_subclassing():
+    """Test application features"""
+
+    # Create an application-specific configuration
+    # Note that ALL OPTIONS MUST HAVE DEFAULTS DEFINED
+    # so that a default config can be constructed with no arguments.
+    class MyAppConfig(AppConfig):
+        network: str = "Network name (e.g. rinkeby, mainnet)"
+        chain_id: int = 0
+
+    # Create Application subclass
+    # Note: other attributes-specific can be added as required
+    class MyApp(Application):
+
+        # Insert other attributes here
+
+        def __init__(self, **data):
+            super().__init__(name="myapp", config_class=MyAppConfig, **data)
+
+    # Make sure temp dir exists
+    testhome.mkdir(parents=True, exist_ok=True)
+
+    #: Instantiate application subclass
+    app = MyApp(homedir=testhome)
+    assert app.homedir == testhome
+    assert app.config.chain_id == 0
+
+    os.remove(app.config.config_file)
+    os.rmdir(testhome)
+
+
+def test_application_processing():
     """Test application startup and shutdown"""
-    app = Application(name="testapp")
+    testhome.mkdir(parents=True, exist_ok=True)
+
+    app = Application(name="myapp", config_class=AppConfig, homedir=testhome)
     assert isinstance(app._shutdown, threading.Event)
-    a = Application(name="myapp")
-    a.startup()
+
+    app = Application(name="myapp", config_class=AppConfig, homedir=testhome)
+    app.startup()
     time.sleep(2)
```

Comment:
Why the wait?

Reply:
This was just a demo.  Waiting 2 seconds because the demo main loop checks for shutdown once per second.  You should see one or two prints pop out from the main thread.

Reply:
Ah, gotcha. Thanks